src/index.ts

```diff
@@ -41,9 +56,31 @@ async function main() {
     await uploadLogs()
   }
 
-//// immediate funcs
+  //// immediate funcs
 
-  function shouldGenerateXcodeproj(): string | false | undefined {
+  function getPlatformInput(input: string): Platform | undefined {
+    const value = core.getInput(input)
+    if (!value) return undefined
+    try {
+      return value as Platform
```

Comment:
@mxcl, on reviewing this code today, I realized this still performs no runtime validation that `value` is a `Platform`. The `catch` clause will never execute.

We could remove this function and just `core.getInput('platform') as Platform` as before, allowing for potential future platforms, or we could validate the input and fail fast with a meaningful `core.error`. Which would you prefer?

Reply:
It's enough that the first function that `getDestination` throws if the platform is not one of the ones we support.

Reply:
Cleaned up in #55.